File: Easy/3658_GCD_of_Odd_and_Even_Sums/Solution.java



public class Solution {
    public int gcdOfOddEvenSums(int n) {
        int oddSum = 0, evenSum = 0;

        for (int i = 1; i <= n*2; i += 2) {
            oddSum += i;
        }

        for (int i = 2; i <= n*2; i += 2) {
            evenSum += i;
        }

        while (evenSum != 0) {
            int temp = oddSum % evenSum;
            oddSum = evenSum;
            evenSum = temp;
        }

        return oddSum;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        System.out.println(solution.gcdOfOddEvenSums(4)); // Output: 4
        
        // test cases 2
        System.out.println(solution.gcdOfOddEvenSums(5)); // Output: 5
        
    }
}

File: Easy/3658_GCD_of_Odd_and_Even_Sums/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.


class Solution {
public:
    int gcd(int a, int b) {
        while (b != 0) {
            int temp = a % b;
            a = b;
            b = temp;
        }
        return a;
    }

    int gcdOfOddEvenSums(int n) {
        int oddSum = 0, evenSum = 0;

        for (int i = 1; i <= n*2; i += 2) {
            oddSum += i;
        }

        for (int i = 2; i <= n*2; i += 2) {
            evenSum += i;
        }

        return gcd(oddSum, evenSum);
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    cout << solution.gcdOfOddEvenSums(4) << endl;

    // test cases 2
    cout << solution.gcdOfOddEvenSums(5) << endl;

    return 0;
}